pennylane/transforms/mitigate.py

```diff
@@ -12,15 +12,346 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Provides transforms for mitigating quantum circuits."""
+from copy import copy
+
 from typing import Any, Dict, Optional, Sequence, Union
 
-from pennylane import QNode, apply
-from pennylane.math import mean, shape
+from pennylane import QNode, apply, adjoint
+from pennylane.math import mean, shape, round
 from pennylane.tape import QuantumTape
 from pennylane.transforms import batch_transform
 
+import pennylane as qml
+
+
+@batch_transform
+def fold_global(circuit, scale_factor):
+    r"""Diffable global circuit folding function as is done in `mitiq.zne.scaling.fold_global <https://mitiq.readthedocs.io/en/v.0.1a2/apidoc.html?highlight=global_folding#mitiq.zne.scaling.fold_global>`_
```

Comment:
```suggestion
    r"""Differentiable global circuit folding function as is done in `mitiq.zne.scaling.fold_global <https://mitiq.readthedocs.io/en/v.0.1a2/apidoc.html#mitiq.zne.scaling.fold_global>`_.
```

Reply:
I think it is good to include a reference to the Mitiq function, however, the first sentence of the docstring could do a better job explaining the purpose of this function 🤔 E.g., compare to the Mitiq example:

> Returns a new circuit obtained by folding the global unitary of the input circuit.